tsl/src/compression/compress_utils.c

```diff
@@ -528,6 +543,14 @@ tsl_decompress_chunk(PG_FUNCTION_ARGS)
 
 	if (uncompressed_chunk->relkind == RELKIND_FOREIGN_TABLE)
 	{
+		/*
+		 * Updating the chunk compression status of the Access Node BEFORE executing remote
+		 * decompression. In the event of failure, the compressed status will be cleared. The
+		 * distributed compression policy will attempt to compress again, which is idempotent, thus
+		 * the metadata are eventually consistent.
```

Comment:
Make it clear that clearing the flag means that it is not certain that the chunk is compressed. You mention it with the flag above, but it is worth repeating here.